driver: sensor: tsl2591: fix unintended sign extension

weijunzhou opened the pull request:
There is a unintended sign extension in the 195 line of tsl2591.c If the result of multiplication is greater than 0x7fffffff, the high bits of cpl would be all 1
Signed-off-by: weijunzhou <20044439@163.com>

ubieda commented on drivers/sensor/ams/tsl2591/tsl2591.c:
```diff
@@ -192,7 +192,7 @@ static int tsl2591_set_threshold(const struct device *dev, enum sensor_attribute
 
 	/* Convert from relative strength of visible light to raw value */
-	cpl = data->atime * data->again;
+	cpl = (uint32_t)data->atime * (uint32_t)data->again;
```
Looks like this was addressed differently in the same file (#76669). Could you verify which way should it go and if both are the same constant, address them both the same way?

weijunzhou replied:
I got it, current way is better than mime! it is perfect, I will close my pr

ubieda replied:
Just to clarify: I did not mean there's nothing to do. The code-snippet modified here needs to be fixed. My suggestion was to just do it the same way in all places.

ubieda reviewed (commented):
Thanks for your contribution!

Please look at the comment left and also change the commit such that there's a line between the body and the sign-off.

Also, looks like you need to rebase to address conflicts.

weijunzhou committed:
driver: sensor: tsl2591: fix unintended sign extension

    There is a unintended sign extension in the 195 line of tsl2591.c. If
    the result of multiplication is greater than 0x7fffffff, the high
    bits of cpl would be all 1.

    Signed-off-by: weijun zhou <20044439@163.com>
